### PR TITLE
update gin library to latest

### LINKED ISSUE #522: Update gin library dependency
Update gin library dependency to latest version.

### PR DESCRIPTION
**Pull request by sancyx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #522 
| License         | Apache 2.0

### What's in this PR?
Update gin library dependency to latest version.

### Why?
Currently used gin module version (1.4.0) is quite old.

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -21,8 +21,9 @@ require (
 	github.com/digitalocean/godo v1.61.0
 	github.com/gin-contrib/cors v0.0.0-20170318125340-cf4846e6a636
 	github.com/gin-contrib/static v0.0.0-20181225054800-cf5e10bbd933
```
Comment (sagikazarmark) [inline go.mod:23]: Can you update these components as well?

File: go.mod [hunk 2]
```diff
@@ -21,8 +21,9 @@ require (
...
-	github.com/gin-gonic/gin v1.4.0
+	github.com/gin-gonic/gin v1.7.2
 	github.com/go-kit/kit v0.10.0
+	github.com/go-playground/validator/v10 v10.4.1 // indirect
```
Comment (sagikazarmark) [inline go.mod:26]: Please run `go mod tidy`

Comment (sancyx) [reply go.mod]: fixed

File: go.mod (hunk 1 above)
Comment (sancyx) [reply go.mod]: updated

**Review by pregnor — APPROVED** [review]

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**